indexer/metadata/simulation/iceprod_tools.py

```diff
@@ -0,0 +1,403 @@
+        if "task" not in job_config["options"]:
+            raise TaskNotFound()
+
+        ret = await self.iceprodv2_rc.request(
+            "GET",
+            f"/datasets/{job_config['options']['dataset_id']}/tasks",
+            {
+                "job_index": job_config["options"]["job"],
+                "keys": "name|task_id|job_id|task_index",
+            },
+        )
+
+        # find matching task
+        task = {}
+        for task in ret.values():
+            if task["name"] == job_config["options"]["task"]:
+                job_config["options"]["task_info"] = task
+                return
+
+        raise TaskNotFound()
+
+    @staticmethod
+    async def _match_outfile_and_add_to_job_config(
+        filepath: str, job_config: dataclasses.Job, job_index: Optional[int],
+    ) -> None:
+        """Add `"task"`, `"job"`, & `"iter"` values to `config["options"]`."""
+        if job_index:  # do we already know what job to look at?
+            job_search: List[int] = [job_index]
+        else:  # otherwise, look at each job from dataset
+            job_search = list(range(job_config["options"]["jobs_submitted"]))
+
+        parser = ExpParser()
+        env = {"parameters": job_config["steering"]["parameters"]}
+
+        def get_path_from_url(f_data: _OutFileData) -> str:
+            url = cast(str, parser.parse(f_data["url"], job_config, env))
+            if "//" not in url:
+                path = url
+            else:
+                path = "/" + url.split("//", 1)[1].split("/", 1)[1]
+            logging.info(f"checking path {path}")
+            return path
+
+        # search each possible file/task
+        possible_outfiles = _IceProdV2Querier._get_outfiles(job_config)
+        for f_data in reversed(possible_outfiles):
+            for job in job_search:
+                for i in range(f_data["iters"]):
+                    logging.info(
+                        f"Searching task: {f_data['task']}, job: {job}, iter: {i}"
+                    )
+                    if get_path_from_url(f_data) == filepath:
+                        logging.info(f"Success on job: {job}, iter: {i}")
+                        job_config["options"].update(
+                            {"task": f_data["task"], "job": job, "iter": i}
+                        )
+                        return
+
+        # if there's no match, at least assign the job_index
+        if job_index:
+            job_config["options"]["job"] = job_index
+        raise OutFileNotFound()
+
+
+# --------------------------------------------------------------------------------------
+# Public Query-Manager interface functions
+
+
+def _get_iceprod_querier(
+    dataset_num: int,
+    iceprodv2_rc: RestClient,
+    iceprodv1_db: pymysql.connections.Connection,
+) -> _IceProdQuerier:
+    if dataset_num in _ICEPROD_V1_DATASET_RANGE:
+        return _IceProdV1Querier(dataset_num, iceprodv1_db)
+    if dataset_num in _ICEPROD_V2_DATASET_RANGE:
```

Comment:
Prefer `elif` for style reasons, because only one of the ifs should be true.

Reply:
Yeah I agree with you on this, but pylint doesn't (R1705). I might as well disable that warning... what would be nice is a robust/modern switch-case statement

For a good laugh: https://pheanex.github.io/pylint/#why-is-this-better

Reply:
I'm actually fine with the no-else-return rule when it's a basic if-return, but less so for multiple conditions.  Though what really does it for me is looking at the logic and wondering if the second `if` could be an `elif`, or if those two conditions could both be true.  I realize the `return` makes it irrelevant, but I see the `if` statements before evaluating the bodies of them.

Also, I suppose in a language like python it doesn't actually matter, but in a compiled language there can be optimizations from knowing the if statements are connected.